### PR TITLE
[Geant] Additional changes for Geant 10.6

### PR DESCRIPTION
**Pull request by mrodozov**
#### PR description:

Additional changes needed for https://github.com/cms-sw/cmsdist/pull/5424 to work. 
See also https://github.com/cms-sw/cmssw/pull/28626

### CODE REVIEW

File: SimG4CMS/Calo/src/CaloSD.cc [hunk 1]
```diff
@@ -321,7 +321,7 @@ bool CaloSD::checkHit() {
       found = true;
     }
   } else if (nCheckedHits > 0) {
-    int minhit = (theHC->entries() > nCheckedHits ? theHC->entries() - nCheckedHits : 0);
+    size_t minhit = (theHC->entries() > nCheckedHits ? theHC->entries() - nCheckedHits : 0);
```
Comment (civanch) [inline SimG4CMS/Calo/src/CaloSD.cc]: @mrodozov , I would move everything to "int":
int nhits = theHC->entries();
int minhit = std::max(nhits - nCheckedHits, 0);

the same everywhere - easy to read and no mix of types anymore.

File: SimG4CMS/Forward/src/BscTest.cc [hunk 2]
```diff
@@ -627,7 +627,7 @@ void BscTest::update(const EndOfEvent* evt) {
     } else {
       varia = 2;
     }  // no MI end:
-    for (int j = 0; j < theCAFI->entries(); j++) {
+    for (size_t j = 0; j < theCAFI->entries(); j++) {
```
Comment (civanch) [inline SimG4CMS/Forward/src/BscTest.cc]: @mrodozov , here also easily everything may be int.

File: SimG4CMS/FP420/plugins/FP420SD.cc [hunk 3]
```diff
@@ -196,7 +196,7 @@ G4bool FP420SD::HitExists() {
 
   //    LogDebug("FP420Sim") << "FP420SD: HCollection=  " << theHC->entries()    <<std::endl;
 
-  for (int j = 0; j < theHC->entries() && !found; j++) {
+  for (size_t j = 0; j < theHC->entries() && !found; j++) {
```
Comment (civanch) [inline SimG4CMS/FP420/plugins/FP420SD.cc]: here also theLH->entries may be once converted to int and used downstream.

File: SimG4CMS/Forward/src/TotemSD.cc [hunk 4]
```diff
@@ -117,7 +117,7 @@ void TotemSD::Initialize(G4HCofThisEvent* HCE) {
 
 void TotemSD::EndOfEvent(G4HCofThisEvent*) {
   // here we loop over transient hits and make them persistent
-  for (int j = 0; j < theHC->entries() && j < 15000; j++) {
+  for (size_t j = 0; j < theHC->entries() && j < 15000; j++) {
```
Comment (civanch) [inline SimG4CMS/Forward/src/TotemSD.cc]: the same

File: SimG4CMS/PPS/src/PPSPixelSD.cc [hunk 5]
```diff
@@ -228,7 +228,7 @@ bool PPSPixelSD::hitExists() {
   //tSliceID_ already exists:
   bool found = false;
 
-  for (int j = 0; j < theHC_->entries() && !found; j++) {
+  for (size_t j = 0; j < theHC_->entries() && !found; j++) {
```
Comment (civanch) [inline SimG4CMS/PPS/src/PPSPixelSD.cc]: the same

File: SimG4CMS/FP420/plugins/FP420Test.cc [hunk 6]
```diff
@@ -954,7 +954,7 @@ void FP420Test::update(const EndOfEvent* evt) {
       varia = 2;
     }  // no MI end:
 
-    for (int j = 0; j < theCAFI->entries(); j++) {
+    for (size_t j = 0; j < theCAFI->entries(); j++) {
```
Comment (civanch) [inline SimG4CMS/FP420/plugins/FP420Test.cc]: @mrodozov , here also easily everything may be
int nhits = theCAFI->entries();

File: SimG4CMS/Forward/src/TimingSD.cc [hunk 7]
```diff
@@ -201,7 +201,7 @@ bool TimingSD::hitExists(const G4Step* aStep) {
   //tSliceID already exists:
 
   bool found = false;
-  for (int j = 0; j < theHC->entries(); ++j) {
+  for (size_t j = 0; j < theHC->entries(); ++j) {
```
Comment (civanch) [inline SimG4CMS/Forward/src/TimingSD.cc]: @mrodozov , this is fine - only one loop, so switch to size_t does not change anything.

**Comment by smuzaffar** [comment]
please test

**Pull request closed without merging**